track_syscalls/binanalyzer/binanalyzer.go

```diff
@@ -0,0 +1,47 @@
+package binanalyzer
+
+import (
+	"debug/elf"
+	"fmt"
+)
+
+type SymbolInfo struct {
+	Name  string
+	Start uint64
+	End   uint64
+}
+
+var Cache []SymbolInfo
+
+func Populate(binaryPath string) error {
+	f, err := elf.Open(binaryPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	symbols, err := f.Symbols()
+	if err != nil {
+		return err
+	}
+
+	for _, sym := range symbols {
+		if sym.Value != 0 {
+			Cache = append(Cache, SymbolInfo{
+				Name:  sym.Name,
+				Start: sym.Value,
+				End:   sym.Value + sym.Size,
+			})
+		}
```

Comment:
What symbols from elf file are you storing here and for what?

Reply:
Since goleash retrieves only the addresses of invoked functions from the stack, we need to translate these addresses into function symbols from the elf binary. Function symbols from the binary come along with package names, which is how goleash can infer the caller dependency. 

note: i don't resolve symbols at runtime during enforcement; instead, I resolve and cache all the function symbols once during initialization, for performance reasons.